scripts/commands/regenerateApiDocs.ts

```diff
@@ -0,0 +1,158 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { readdir } from "fs/promises";
+
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import { $ } from "zx";
+import fs from "fs";
+
+import { Pkg } from "../lib/api/Pkg";
+import { zxMain } from "../lib/zx";
+
+interface Arguments {
+  [x: string]: unknown;
+  package?: string;
+  historical: boolean;
+}
+
+const readArgs = (): Arguments => {
+  return yargs(hideBin(process.argv))
+    .version(false)
+    .option("package", {
+      alias: "p",
+      type: "string",
+      choices: Pkg.VALID_NAMES,
+      demandOption: false,
+      description: "Which package to update",
+    })
+    .option("historical", {
+      type: "boolean",
+      default: false,
+      description: "Regenerate prior releases?",
+    })
+    .parseSync();
+};
+
+zxMain(async () => {
+  const args = readArgs();
+
+  const initialStatus = await gitStatus();
+  if (initialStatus !== "") {
+    console.error(`
+        Repository must have clean Git state when calling
+        \`npm run regenerate-apis\`. 'git status' returned:\n\n${initialStatus}`);
+    process.exit(1);
+  }
+
+  const currentBranch = await gitBranch();
+  if (currentBranch == "main\n") {
+    console.error(`
+        Please create a dedicated branch to regenerate the API docs correctly.
+        Use 'git checkout -b <name-branch>' to continue`);
+    process.exit(1);
+  }
+
+  const results = new Map<string, string[]>();
+  for (const pkgName of Pkg.VALID_NAMES) {
+    if (args.package && pkgName != args.package) {
+      continue;
+    }
+
+    const pkgVersions = await getPackageVersions(pkgName, args.historical);
+    const result = await regenerateVersions(pkgName, pkgVersions);
+    results.set(pkgName, result);
+  }
+
+  console.log("");
+  results.forEach((result: string[], pkgName: string) => {
+    console.log(`Regeneration of ${pkgName}:`);
+    result.forEach((msg) => console.error(msg));
+    console.log("");
+  });
+});
+
+async function regenerateVersions(
+  pkgName: string,
+  versions: string[],
+): Promise<string[]> {
+  const results: string[] = [];
+
+  for (const [index, version] of versions.entries()) {
+    try {
+      if (index == versions.length - 1) {
+        // The last version is always the latest stable minor release
```

Comment:
More clear would be to take 3 args in the function: `currentVersion` and `historicalVersions`. That's a type-driven API - the type system enforces your "invariant" for you.